RecoLocalTracker/SubCollectionProducers/src/JetCoreClusterSplitter.cc

```diff
@@ -126,18 +151,18 @@ void JetCoreClusterSplitter::produce(edm::Event& iEvent, const edm::EventSetup&
           if (Geom::deltaR(jetDir, clusterDir) < deltaR_) {
             // check if the cluster has to be splitted
 
-            bool isEndCap = (std::abs(cPos.z()) > 30.f);  // FIXME: check detID instead!
+            LocalVector jetDirLocal = det->surface().toLocal(jetDir);
+            float jetTanAlpha = jetDirLocal.x() / jetDirLocal.z();
+            float jetTanBeta = jetDirLocal.y() / jetDirLocal.z();
             float jetZOverRho = jet.momentum().Z() / jet.momentum().Rho();
-            if (isEndCap)
-              jetZOverRho = jet.momentum().Rho() / jet.momentum().Z();
-            float expSizeY = std::sqrt((1.3f * 1.3f) + (1.9f * 1.9f) * jetZOverRho * jetZOverRho);
+            float expSizeX = expSizeXAtLorentzAngleIncidence_ +
+                             std::abs(expSizeXDeltaPerTanAlpha_ * (jetTanAlpha - tanLorentzAngle));
+            float expSizeY = std::sqrt((expSizeYAtNormalIncidence_ * expSizeYAtNormalIncidence_) +
+                                       thickness_ * thickness_ / (pitchY_ * pitchY_) * jetTanBeta * jetTanBeta);
+            if (expSizeX < 1.f)
+              expSizeX = 1.f;
             if (expSizeY < 1.f)
               expSizeY = 1.f;
-            float expSizeX = 1.5f;
-            if (isEndCap) {
```

Comment:
why the endcap special treatment is not applied anymore ?

Reply:
In general, the treatment that was applied in the case of a module from the endcaps should now be covered by computing everything in the local frame of the module.

However, it is also true that I initially forgot to update the definition of `jetZOverRho`, such that its value was wrong in the endcaps. I've now fixed this, and I've rerun a step3 + validation to check that everything it's still ok.

(As a general comment, the value of `jetZOverRho`  in the endcaps now is slightly different from upstream CMSSW, due to the turbine-like geometry that previously was not not taken into account.)